scriptmodules/supplementary/launchingimages.sh

```diff
@@ -0,0 +1,413 @@
+        return 1
+    fi
+}
+
+function _get_config_file_launchingimages() {
+    local file_list=$(find "$md_inst" -type f -name '*.cfg' ! -name '.current.cfg' | sort | xargs)
+    if [[ -z "$file_list" ]]; then
+        printMsgs "dialog" "There's no config file saved."
+        return 1
+    fi
+    _dialog_menu_launchingimages "Choose the file" $file_list # XXX: no quotes surrounding $file_list is mandatory!
+    return $?
+}
+
+function _load_config_launchingimages() {
+    echo "$(loadModuleConfig \
+        'theme=' \
+        'extension=png' \
+        'show_timeout=5' \
+        'loading_text=NOW LOADING' \
+        'press_button_text=PRESS A BUTTON TO CONFIGURE LAUNCH OPTIONS' \
+        'loading_text_color=white' \
+        'press_button_text_color=gray50' \
+        'no_ask=' \
+        'no_logo=' \
+        'solid_bg_color=' \
+        'system=' \
+        'logo_belt=' \
+    )"
+}
+
+function _settings_launchingimages() {
+    local cmd=(dialog --backtitle "$__backtitle" --title " SETTINGS " --menu "runcommand launching images generation settings." 22 86 16)
+    local options
+    local choice
+    local config_file
+
+    iniConfig ' = ' '"' "$md_inst/.current.cfg"
+
+    while true; do
+        eval $(_load_config_launchingimages)
+
+        options=( 
+            config_file "$(
+               [[ "$config_file" =~ $theme\.cfg$ ]] && echo "$(basename "$config_file")"
+            )"
+            theme "$theme"
+            system "$(
+                if [[ -z "$system" ]]; then
+                    echo "all systems in es_systems.cfg"
+                else
+                    echo "$system" | cut -d' ' -f2
+                fi
+            )"
+            extension ".$extension"
+            loading_text "\"$loading_text\""
+            press_button_text "\"$press_button_text\""
+            loading_text_color "$loading_text_color"
+            press_button_text_color "$press_button_text_color"
+            show_timeout "$( [[ -n "$no_ask" ]] && echo "don't show (see no_ask)" || echo "$show_timeout seconds")"
+            no_ask "$( [[ -n "$no_ask" ]] && echo true || echo false)"
+            no_logo "$( [[ -n "$no_logo" ]] && echo true || echo false)"
+            logo_belt "$( [[ -n "$logo_belt" ]] && echo true || echo false)"
+            solid_bg_color "$(
+                if [[ -z "$solid_bg_color" ]]; then
+                    echo false
+                elif [[ -z "$(echo "$solid_bg_color" | cut -s -d' ' -f2)" ]]; then
+                    echo "get from the theme"
+                else
+                    echo "$solid_bg_color" | cut -s -d' ' -f2
+                fi
+            )"
+        )
+        choice=$("${cmd[@]}" "${options[@]}" 2>&1 >/dev/tty)
+
+        [[ -z "$choice" ]] && break
+
+        if [[ "$choice" = "config_file" ]]; then
+            config_file=$(_manage_config_file_launchingimages)
+            continue
+        fi
+
+        eval iniSet "$choice" \"$(_set_${choice}_launchingimages)\"
```

Comment:
oh. you are returning a variable name in a called function - wouldn't it be best if the functions just echoed the variable value - this seems more logical.